File: src/.NET/fix40/Logon.cs
namespace QuickFix40
{

  public class Logon : Message
  {
    public Logon() : base(MsgType()) {}
    static QuickFix.MsgType MsgType() { return new QuickFix.MsgType("A"); }

    public Logon(
      QuickFix.EncryptMethod aEncryptMethod,
      QuickFix.HeartBtInt aHeartBtInt )
    : base(MsgType()) {
      set(aEncryptMethod);
      set(aHeartBtInt);
    }

    public void set(QuickFix.EncryptMethod value)
    { setField(value); }
    public QuickFix.EncryptMethod get(QuickFix.EncryptMethod  value)
    { getField(value); return value; }
    public QuickFix.EncryptMethod getEncryptMethod()
    { QuickFix.EncryptMethod value = new QuickFix.EncryptMethod();
      getField(value); return value; }
    public bool isSet(QuickFix.EncryptMethod field)
    { return isSetField(field); }
    public bool isSetEncryptMethod()
    { return isSetField(98); }

    public void set(QuickFix.HeartBtInt value)
    { setField(value); }
    public QuickFix.HeartBtInt get(QuickFix.HeartBtInt  value)
    { getField(value); return value; }
    public QuickFix.HeartBtInt getHeartBtInt()
    { QuickFix.HeartBtInt value = new QuickFix.HeartBtInt();
      getField(value); return value; }
    public bool isSet(QuickFix.HeartBtInt field)
    { return isSetField(field); }
    public bool isSetHeartBtInt()
    { return isSetField(108); }

    public void set(QuickFix.RawDataLength value)
    { setField(value); }
    public QuickFix.RawDataLength get(QuickFix.RawDataLength  value)
    { getField(value); return value; }
    public QuickFix.RawDataLength getRawDataLength()
    { QuickFix.RawDataLength value = new QuickFix.RawDataLength();
      getField(value); return value; }
    public bool isSet(QuickFix.RawDataLength field)
    { return isSetField(field); }
    public bool isSetRawDataLength()
    { return isSetField(95); }

    public void set(QuickFix.RawData value)
    { setField(value); }
    public QuickFix.RawData get(QuickFix.RawData  value)
    { getField(value); return value; }
    public QuickFix.RawData getRawData()
    { QuickFix.RawData value = new QuickFix.RawData();
      getField(value); return value; }
    public bool isSet(QuickFix.RawData field)
    { return isSetField(field); }
    public bool isSetRawData()
    { return isSetField(96); }

  };

}

File: src/java/src/quickfix/fix40/Logon.java
package quickfix.fix40;
import quickfix.FieldNotFound;
import quickfix.Group;
import quickfix.field.*;

public class Logon extends Message
{
  public Logon()
  {
    getHeader().setField(new MsgType("A"));
  }
  public Logon(
    quickfix.field.EncryptMethod aEncryptMethod,
    quickfix.field.HeartBtInt aHeartBtInt ) {

      getHeader().setField(new MsgType("A"));
      set(aEncryptMethod);
      set(aHeartBtInt);
  }

  public void set(quickfix.field.EncryptMethod value)
  { setField(value); }
  public quickfix.field.EncryptMethod get(quickfix.field.EncryptMethod  value)
    throws FieldNotFound
  { getField(value); return value; }
  public quickfix.field.EncryptMethod getEncryptMethod() throws FieldNotFound
  { quickfix.field.EncryptMethod value = new quickfix.field.EncryptMethod();
    getField(value); return value; }
  public boolean isSet(quickfix.field.EncryptMethod field)
  { return isSetField(field); }
  public boolean isSetEncryptMethod()
  { return isSetField(98); }
  public void set(quickfix.field.HeartBtInt value)
  { setField(value); }
  public quickfix.field.HeartBtInt get(quickfix.field.HeartBtInt  value)
    throws FieldNotFound
  { getField(value); return value; }
  public quickfix.field.HeartBtInt getHeartBtInt() throws FieldNotFound
  { quickfix.field.HeartBtInt value = new quickfix.field.HeartBtInt();
    getField(value); return value; }
  public boolean isSet(quickfix.field.HeartBtInt field)
  { return isSetField(field); }
  public boolean isSetHeartBtInt()
  { return isSetField(108); }
  public void set(quickfix.field.RawDataLength value)
  { setField(value); }
  public quickfix.field.RawDataLength get(quickfix.field.RawDataLength  value)
    throws FieldNotFound
  { getField(value); return value; }
  public quickfix.field.RawDataLength getRawDataLength() throws FieldNotFound
  { quickfix.field.RawDataLength value = new quickfix.field.RawDataLength();
    getField(value); return value; }
  public boolean isSet(quickfix.field.RawDataLength field)
  { return isSetField(field); }
  public boolean isSetRawDataLength()
  { return isSetField(95); }
  public void set(quickfix.field.RawData value)
  { setField(value); }
  public quickfix.field.RawData get(quickfix.field.RawData  value)
    throws FieldNotFound
  { getField(value); return value; }
  public quickfix.field.RawData getRawData() throws FieldNotFound
  { quickfix.field.RawData value = new quickfix.field.RawData();
    getField(value); return value; }
  public boolean isSet(quickfix.field.RawData field)
  { return isSetField(field); }
  public boolean isSetRawData()
  { return isSetField(96); }
}

